google/generativeai/types/content_types.py

```diff
@@ -723,20 +746,36 @@ def _make_tool(tool: ToolType) -> Tool:
             code_execution = tool.code_execution
         else:
             code_execution = None
-        return Tool(function_declarations=tool.function_declarations, code_execution=code_execution)
+        return Tool(
+            function_declarations=tool.function_declarations,
+            google_search_retrieval=tool.google_search_retrieval,
+            code_execution=code_execution,
+        )
     elif isinstance(tool, dict):
```

Comment:
In this elif is where you want to convert that enum Mode (if something was passed).

You should probably do that in a stand alone function. `_make_google_search_retrieval`?